app/models/page/DashboardPage.scala

```diff
@@ -16,111 +16,251 @@
 
 package models.page
 
-import models.ApplicationData.ApplicationStatus
-import models.ApplicationData.ApplicationStatus.ApplicationStatus
-import models.page.DashboardPage.ProgressStepVisibility
-import models.{ ApplicationData, CachedData, Progress }
+import connectors.{ AllocationExchangeObjects, ExchangeObjects }
+import models.page.DashboardPage.Flags.{ ProgressInactive, _ }
+import models.{ CachedData, Progress }
+import org.joda.time.LocalDate
 import play.api.i18n.Lang
 import play.api.mvc.RequestHeader
-import security.RoleUtils
-import security.Roles.DisplayOnlineTestSectionRole
+import security.RoleUtils._
+import security.Roles._
 
-// format: OFF
 case class DashboardPage(firstStepVisibility: ProgressStepVisibility,
                          secondStepVisibility: ProgressStepVisibility,
                          thirdStepVisibility: ProgressStepVisibility,
-                         fourthStepVisibility: ProgressStepVisibility)
-
-// format: ON
+                         fourthStepVisibility: ProgressStepVisibility,
+                         isApplicationSubmittedAndNotWithdrawn: Boolean,
+                         isApplicationInProgressAndNotWithdrawn: Boolean,
+                         isApplicationWithdrawn: Boolean,
+                         isApplicationCreatedOrInProgress: Boolean,
+                         isUserWithNoApplication: Boolean,
+                         isFirstStepVisible: String,
```

Comment:
Well... that was just a way to move the original "setVisibility" away from the page and into the model but I don't think it was the best way. setVisibility was just returning a "active"/"disable" value based on the input step. I removed from the list of arguments and create a method (activateByStep) which should make more sense.